src/components/task-list/task-list.component.js

```diff
@@ -43,24 +59,42 @@ const TaskList = (props) => {
               <li
                 className={task.completed ? styles.completedTask : styles.task}
                 key={task.id}
+                id={task.id}
+                name={task.name}
               >
-                <h4>Task Name:</h4>
-                <h5>{task.name}</h5>
-                <h4>Due Date:</h4>
-                <h5>{task.dueDate}</h5>
-                <h4>Priority:</h4>
-                <h5>{task.priority}</h5>
+                <h4 className={styles.smallMarginRight}>Task Name:</h4>
+                <h5 className={styles.marginRight}>{task.name}</h5>
+                <h4 className={styles.smallMarginRight}>Due Date:</h4>
+                <h5 className={styles.marginRight}>{task.dueDate}</h5>
+                <h4 className={styles.smallMarginRight}>Priority:</h4>
+                <h5 className={styles.marginRight}>{task.priority}</h5>
                 <input
                   id={task.id}
-                  role="button"
+                  role='button'
                   checked={task.completed}
                   hover-message={
-                    task.completed ? "Mark as Incomplete" : "Mark as Complete"
+                    task.completed ? 'Mark as Incomplete' : 'Mark as Complete'
                   }
                   className={styles.checkbox}
-                  type="checkbox"
+                  type='checkbox'
                   onChange={clickCheckbox}
+                  style={{ cursor: 'pointer' }}
                 ></input>
+                <div
+                  onClick={openModal}
```

Comment:
you could do something like `onClick={() => openModal(task)}`

then your openModal can be defined as (forget about the event at this point):
```javascript
  const openModal = (task) => {
    setShowModal(true);
    setSelectTask(task); // now the selectTask is actually holding a task and no an event
  };
```

then your modal should only deal with a task

Reply:
2d01ae9b55c1df240310c64785203b7e191b5ecb Changed as advised. Makes so much more sense like this and doesn`t run to the problem of not selecting the correct element plus of being much cleaner code.  Thank you!

Reply:
🙌🏽  perfect!! hope all the suggestions are making sense!